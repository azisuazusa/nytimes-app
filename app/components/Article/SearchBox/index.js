/*
 * @flow
 */

import React, { Component } from 'react';
import { Input, Grid, Col } from 'native-base';
import CommonPicker from '../../Common/CommonPicker';
import constants from '../../../configs/constants';

type Props = {
  textValue: string,
  pickerValue: string,
  onValueChange: (value: string) => void,
  onChangeText: (value: string) => void
};

export default class SearchBox extends Component<Props> {
  render() {
    const { textValue, pickerValue, onValueChange, onChangeText } = this.props;
    return (
      <Grid>
        <Col>
          <Input
            placeholder="Search..."
            value={textValue}
            onChangeText={onChangeText} />
        </Col>
        <Col>
          <CommonPicker
            data={constants.sortArticle}
            onValueChange={onValueChange}
            value={pickerValue} />
        </Col>
      </Grid>
    );
  }
}
