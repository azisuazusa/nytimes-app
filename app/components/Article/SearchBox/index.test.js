import React from 'react';
import renderer from 'react-test-renderer';
import SearchBox from '../SearchBox';

it('SearchBox renders correctly', () => {
  const tree = renderer.create(
    <SearchBox
      onValueChange={() => null}
      onChangeText={() => null}
      textValue={'New York Times'}
      pickerValue={'newest'} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
