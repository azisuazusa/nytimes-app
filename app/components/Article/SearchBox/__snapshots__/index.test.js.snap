// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      onChangeText={[Function]}
      placeholder="Search..."
      placeholderTextColor="#575757"
      style={
        Object {
          "color": "#000",
          "flex": 1,
          "fontSize": 17,
          "height": 50,
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value="New York Times"
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": null,
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          Newest
        </Text>
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        transparent={false}
        visible={false}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
              "height": 1334,
            }
          }
        >
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F8F8F8",
                  "borderBottomColor": "#a7a6ab",
                  "borderBottomWidth": 0.5,
                  "elevation": 3,
                  "flexDirection": "row",
                  "height": 64,
                  "justifyContent": "center",
                  "left": 0,
                  "paddingLeft": 6,
                  "paddingRight": 10,
                  "paddingTop": 18,
                  "right": 0,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": null,
                      "backgroundColor": "transparent",
                      "borderBottomWidth": null,
                      "borderColor": null,
                      "borderLeftWidth": null,
                      "borderRadius": 5,
                      "borderRightWidth": null,
                      "borderTopWidth": null,
                      "borderWidth": undefined,
                      "elevation": 0,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "space-between",
                      "marginLeft": 3,
                      "opacity": 1,
                      "paddingBottom": 6,
                      "paddingLeft": 4,
                      "paddingRight": 6,
                      "paddingTop": 6,
                      "shadowColor": null,
                      "shadowOffset": null,
                      "shadowOpacity": null,
                      "shadowRadius": null,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#007aff",
                        "fontFamily": "System",
                        "fontSize": 17,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 2,
                        "paddingRight": undefined,
                        "top": 1,
                      }
                    }
                    uppercase={false}
                  >
                    Back
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "fontWeight": "700",
                      "marginLeft": undefined,
                      "paddingLeft": 4,
                      "paddingTop": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Select One
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              />
            </View>
          </View>
          <RCTScrollView
            data={
              Array [
                <Item
                  label="Sort by..."
                  value=""
                />,
                <Item
                  label="Newest"
                  value="newest"
                />,
                <Item
                  label="Oldest"
                  value="oldest"
                />,
              ]
            }
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={null}
                >
                  <View
                    button={true}
                    onPress={[Function]}
                    selected={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderBottomWidth": 0.5,
                        "borderColor": "#c9c9c9",
                        "flexDirection": "row",
                        "marginLeft": 16,
                        "paddingBottom": 13,
                        "paddingRight": 16,
                        "paddingTop": 13,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "alignSelf": "center",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          }
                        }
                        uppercase={false}
                      >
                        Sort by...
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "flex-end",
                          "alignSelf": "center",
                          "flex": 0.28,
                          "paddingBottom": null,
                          "paddingLeft": null,
                          "paddingRight": null,
                          "paddingTop": null,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        isTVSelectable={true}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignSelf": null,
                            "opacity": 1,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={null}
                >
                  <View
                    button={true}
                    onPress={[Function]}
                    selected={true}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderBottomWidth": 0.5,
                        "borderColor": "#c9c9c9",
                        "flexDirection": "row",
                        "marginLeft": 16,
                        "paddingBottom": 13,
                        "paddingRight": 16,
                        "paddingTop": 13,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "alignSelf": "center",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#007aff",
                            "fontFamily": "System",
                            "fontSize": 16,
                          }
                        }
                        uppercase={false}
                      >
                        Newest
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "flex-end",
                          "alignSelf": "center",
                          "flex": 0.28,
                          "paddingBottom": null,
                          "paddingLeft": null,
                          "paddingRight": null,
                          "paddingTop": null,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        isTVSelectable={true}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignSelf": null,
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 12,
                              },
                              Object {
                                "color": "#007aff",
                                "fontSize": 25,
                                "height": 20,
                                "lineHeight": 25,
                              },
                              Object {
                                "fontFamily": "Ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={null}
                >
                  <View
                    button={true}
                    onPress={[Function]}
                    selected={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderBottomWidth": 0.5,
                        "borderColor": "#c9c9c9",
                        "flexDirection": "row",
                        "marginLeft": 16,
                        "paddingBottom": 13,
                        "paddingRight": 16,
                        "paddingTop": 13,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "alignSelf": "center",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          }
                        }
                        uppercase={false}
                      >
                        Oldest
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "flex-end",
                          "alignSelf": "center",
                          "flex": 0.28,
                          "paddingBottom": null,
                          "paddingLeft": null,
                          "paddingRight": null,
                          "paddingTop": null,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        isTVSelectable={true}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignSelf": null,
                            "opacity": 1,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
